### ISSUE TITLE
Completions for variables include the `var.` (unless you don't type the dot)

### ISSUE THREAD

**Issue by awilkins** [hashicorp/vscode-terraform#1624]
### Extension Version

v2.29.0

### VS Code Version

Version: 1.84.1
Commit: 2b35e1e6d88f1ce073683991d1eff5284a32690f
Date: 2023-11-06T12:35:31.181Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 5.15.0-88-generic

### Operating System

Ubuntu 20.04

### Terraform Version

Terraform v1.6.3 on linux_amd64

### Steps to Reproduce

1. Create a variable
2. Use this variable in a resource block - type `var.`, receive a list of completions
3. Select the variable in the completions list

### Expected Behavior

Completion finishes variable

ie you get

```
  var.my_variable
```

### Actual Behavior

You get

```
  var.var.my_variable
```

### Workarounds

This only happens when you type the dot after var ; presumably at this point, the completions should change from

```
var.
     | var.my_variable |
```

to

```
var.
      | my_variable |
```

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by radeksimko** [comment]
Hi @awilkins 
I wasn't able to reproduce this one.

![2023-11-09 13 12 46](https://github.com/hashicorp/vscode-terraform/assets/287584/500a434d-b899-487b-854f-975789d04f81)

Completion can be very context-dependent though and I can imagine there is a bug somewhere. Can you provide a larger snippet of code?

**Comment by awilkins** [comment]
I can confirm that the simple test above also works fine for me - what triggers it is when the variable is nested in an array

Tried it in nested blocks as well, that didn't cause the issue.

![Peek 2023-11-09 14-09](https://github.com/hashicorp/vscode-terraform/assets/368399/99343d78-e149-4373-ba74-cd5134736a56)

**Comment by radeksimko** [comment]
Thanks @awilkins I was able to reproduce with the following snippet and can confirm this is a bug that needs fixing.

```hcl
variable "my_variable" {
}

resource "random_pet" "name" {
  depends_on = [ var. ]
}
```

**Comment by dbanck** [comment]
We just released a new version [2.29.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.29.1). This fixes the reported bug. The update should appear automatically in VS Code.

Thank you for the report @awilkins
